### PR TITLE
Handle segmenting everywhere in data_tools

### PR DESCRIPTION
**Pull request by as2388**
Updates the rest of data_tools to handle segmenting (including setting and getting code schemes, users).

Most of the changes are to firebase_client_wrapper.py, with minimal changes to other users of this library. However, I found lots of duplicated implementations of fcw code in the other data tools scripts, so updated these to use the library functions instead.

cc @IsaackMwenda

### CODE REVIEW

**Comment by lukechurch** [comment]
Hi @as2388 - for future reference I would really have preferred this as multiple PRs, separating the incidental changes from the core changes to fcw

File: data_tools/firebase_client_wrapper.py [hunk 1]
```diff
@@ -138,44 +232,71 @@ def set_segment_messages_content_batch(dataset_id, messages, segment_index=None,
...
 
     set_segment_messages_content_batch(
         dataset_id, next_batch_to_write, segment_index=get_segment_count(dataset_id), batch_size=batch_size)
 
+
+def delete_segment(segment_id):
+    # Delete code schemes
+    segment_code_schemes_path = f"datasets/{segment_id}/code_schemes"
+    print(f"Deleting {segment_code_schemes_path}...")
+    _delete_collection(client.collection(segment_code_schemes_path), 10)
```
Comment (lukechurch) [inline data_tools/firebase_client_wrapper.py]: This seems a pretty small batch size? I don't recall if delete batches have to be smaller?

File: data_tools/firebase_client_wrapper.py [hunk 2]
```diff
@@ -9,70 +10,154 @@
...
+
 def init_client(crypto_token_path):
     global client
     cred = credentials.Certificate(crypto_token_path)
     firebase_admin.initialize_app(cred)
     client = firestore.client()
 
-def get_dataset_ids():
+
+def id_for_segment(dataset_id, segment_index=None):
```
[the remark this answers is not in the document]
Comment (as2388) [reply data_tools/firebase_client_wrapper.py]: It's used quite intensively by the other functions in fcw, so I think this is the right place.

Appreciate that may not have been obvious when reading from the top of the file

File: data_tools/firebase_client_wrapper.py [hunk 3]
```diff
@@ -9,70 +10,154 @@
...
     return ids
 
-def get_dataset(id):
-    return client.document(u'datasets/{}'.format(id)).get()
 
-def get_dataset_ref(id):
-    return client.document(u'datasets/{}'.format(id))
+def get_dataset_ids():
+    # Get the dataset ids by retrieving all the segment ids then removing those that are not the first segment.
+    dataset_ids = set(get_segment_ids())
```
[the remark this answers is not in the document]
Comment (as2388) [reply data_tools/firebase_client_wrapper.py]: added

File: data_tools/firebase_client_wrapper.py [hunk 4]
```diff
@@ -9,70 +10,154 @@
...
+        segment_count = get_segment_count(dataset_id)
+        if segment_count is not None and segment_count > 1:
+            for segment_index in range(2, segment_count + 1):
+                dataset_ids.remove(id_for_segment(dataset_id, segment_index))
+
+    return dataset_ids
+
+
+def get_segment(segment_id):
+    return client.document(u'datasets/{}'.format(segment_id)).get()
```
[the remark this answers is not in the document]
Comment (as2388) [reply data_tools/firebase_client_wrapper.py]: that's much nicer, thanks

File: data_tools/firebase_client_wrapper.py (hunk 1 above)
Comment (as2388) [reply data_tools/firebase_client_wrapper.py]: I believe they can be the max size, I was just following your lead of 10 at a time.

But if you're happy with an increase I will bump to the maximum

File: data_tools/firebase_client_wrapper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (as2388) [reply data_tools/firebase_client_wrapper.py]: This was a bit more substantial than a usual PR edit, so opened a new PR https://github.com/AfricasVoices/CodaV2/pull/231 with that change so you can review

**Pull request merged**